application/indexer/value.go

```diff
@@ -93,53 +99,74 @@ func (vi *ValueIndex) Drop(txn *badger.Txn, utxoID []byte) error {
 	return utils.DeleteValue(txn, key)
 }
 
-func (vi *ValueIndex) GetValueForOwner(txn *badger.Txn, owner *objs.Owner, minValue *uint256.Uint256, exclude [][]byte) ([][]byte, *uint256.Uint256, error) {
-	exclusionSet := make(map[[constants.HashLen]byte]bool)
-	var hsh [constants.HashLen]byte
-	for j := 0; j < len(exclude); j++ {
-		ID := utils.CopySlice(exclude[j])
-		copy(hsh[:], utils.CopySlice(ID))
-		exclusionSet[hsh] = true
-	}
-	result := [][]byte{}
-	valueCount := uint256.Zero()
-	opts := badger.DefaultIteratorOptions
-	prefix := vi.prefix()
+func (vi *ValueIndex) GetValueForOwner(txn *badger.Txn, owner *objs.Owner, minValue *uint256.Uint256, excludeFn func([]byte) (bool, error), maxCount int, lastKey []byte) ([][]byte, *uint256.Uint256, []byte, error) {
 	ownerBytes, err := owner.MarshalBinary()
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
+
+	prefix := vi.prefix()
 	prefix = append(prefix, ownerBytes...)
-	prefixLen := len(prefix)
+	opts := badger.DefaultIteratorOptions
 	opts.Prefix = prefix
 	iter := txn.NewIterator(opts)
 	defer iter.Close()
-	for iter.Seek(prefix); iter.ValidForPrefix(prefix); iter.Next() {
+
+	result := [][]byte{}
+	totalValue := uint256.Zero()
+	prefixLen := len(prefix)
+
+	if lastKey != nil {
```

Comment:
Can this be moved such that the seek offset is calculated before the for loop such that for loop semantics may be preserved? Not a hard requirement, but may be easier to read due to standard pattern.